packages/common/driver-definitions/src/storage.ts

```diff
@@ -161,6 +161,11 @@ export interface IDocumentStorageService extends Partial<IDisposable> {
 	// eslint-disable-next-line @rushstack/no-new-null
 	getSnapshotTree(version?: IVersion, scenarioName?: string): Promise<ISnapshotTree | null>;
 
+	getSnapshot?(
+		version?: IVersion,
+		snapshotFetchOptions?: ISnapshotFetchOptions,
+	): Promise<ISnapshot | undefined>;
```

Comment:
Do we ever get undefined nowadays? I'd think we either get exception or result. Might be worth cleaning all these paths that are no longer possible (if that's the case)

Reply:
We get undefined, for example, in case where let's say there is no snapshot URL in the driver to connect to fetch the snapshot. In that case, we supply undefined rather than throwing an exception. 
Now, we can say that that if no snapshot, is there than we will be throwing an exception.

Reply:
Removed undefined.